templates/accounts.go

```diff
@@ -26,58 +26,126 @@ import (
 //
-// This template accepts a list of public keys and a code argument, both of which are optional.
+// This template accepts an optional list of public keys.
 //
 // The final argument is the address of the account that will pay the account creation fee.
 // This account is added as a transaction authorizer and therefore must sign the resulting transaction.
-func CreateAccount(accountKeys []*flow.AccountKey, code []byte, payer flow.Address) *flow.Transaction {
+func CreateAccount(accountKeys []*flow.AccountKey, payer flow.Address) *flow.Transaction {
 	publicKeys := make([]cadence.Value, len(accountKeys))
 
 	for i, accountKey := range accountKeys {
 		publicKeys[i] = bytesToCadenceArray(accountKey.Encode())
 	}
 
 	cadencePublicKeys := cadence.NewArray(publicKeys)
-	cadenceCode := bytesToCadenceArray(code)
 
 	return flow.NewTransaction().
 		SetScript([]byte(createAccountTemplate)).
 		AddAuthorizer(payer).
+		AddRawArgument(jsoncdc.MustEncode(cadencePublicKeys))
+}
+
+// CreateAccountWithContracts generates a transactions that creates a new account.
+//
+// This template accepts a list of public keys and a contracts argument, both of which are optional.
+//
+// The contracts argument is a dictionary of *contract name*: *contract code (in bytes)*.
+// All of the contracts will be deployed to the account.
+//
+// The final argument is the address of the account that will pay the account creation fee.
+// This account is added as a transaction authorizer and therefore must sign the resulting transaction.
+func CreateAccountWithContracts(accountKeys []*flow.AccountKey, contracts map[string][]byte, payer flow.Address) *flow.Transaction {
+	publicKeys := make([]cadence.Value, len(accountKeys))
+
+	for i, accountKey := range accountKeys {
+		publicKeys[i] = bytesToCadenceArray(accountKey.Encode())
+	}
+
+	contractKeyPairs := make([]cadence.KeyValuePair, len(contracts))
+	i := 0
+	for s, bytes := range contracts {
+		contractKeyPairs[i] = cadence.KeyValuePair{
+			Key:   cadence.NewString(s),
+			Value: bytesToCadenceArray(bytes),
+		}
+		i++
+	}
+
+	cadencePublicKeys := cadence.NewArray(publicKeys)
+	cadenceContracts := cadence.NewDictionary(contractKeyPairs)
+
+	return flow.NewTransaction().
+		SetScript([]byte(createAccountWitContractsTemplate)).
+		AddAuthorizer(payer).
 		AddRawArgument(jsoncdc.MustEncode(cadencePublicKeys)).
-		AddRawArgument(jsoncdc.MustEncode(cadenceCode))
+		AddRawArgument(jsoncdc.MustEncode(cadenceContracts))
 }
 
 const updateAccountCodeTemplate = `
-transaction(code: [UInt8]) {
+transaction(name: String, code: [UInt8]) {
   prepare(signer: AuthAccount) {
-	signer.setCode(code)
+	signer.contracts.update__experimental(name: name, code: code)
   }
 }
 `
 
 // UpdateAccountCode generates a transaction that updates the code deployed at an account.
-func UpdateAccountCode(address flow.Address, code []byte) *flow.Transaction {
+func UpdateAccountCode(address flow.Address, name string, code []byte) *flow.Transaction {
```

Comment:
```suggestion
func UpdateAccountContract(address flow.Address, name string, code []byte) *flow.Transaction {
```

In general I like the idea of retiring the term "code" in our SDKs -- I think it makes a lot more sense to refer to specific contracts on an account.